src/components/react-hook-form/custom-form-provider.tsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { createContext, PropsWithChildren, useContext } from 'react';
+import { FormProvider, useFormContext, UseFormReturn } from 'react-hook-form';
+import * as yup from 'yup';
+
+type CustomFormContextProps = {
+    removeOptional?: boolean;
+    validationSchema: yup.AnySchema;
+};
+
+type MergedFormContextProps = UseFormReturn<any> & CustomFormContextProps;
+
+type CustomFormProviderProps = PropsWithChildren<MergedFormContextProps>;
+
+const CustomFormContext = createContext<CustomFormContextProps>({
+    removeOptional: false,
+    validationSchema: yup.object(),
+});
+
+const CustomFormProvider = (props: CustomFormProviderProps) => {
+    const { validationSchema, removeOptional, children, ...formMethods } =
+        props;
+
+    return (
+        <FormProvider {...formMethods}>
+            <CustomFormContext.Provider
+                value={{
+                    validationSchema: validationSchema,
+                    removeOptional: removeOptional,
+                }}
+            >
+                {children}
+            </CustomFormContext.Provider>
+        </FormProvider>
+    );
+};
+
+export default CustomFormProvider;
+
+export const useCustomFormContext = (): MergedFormContextProps => {
```

Comment:
I'm not sure it's a good idea for a 3 lines hook, especially in this case, both are very linked.
If you check how RHF exports its types, you can see that both FormProvider and useFormContext are exported from the same .d.ts file.
For me sometimes it's better to have a 50 line file instead of lot of 5 10 lines files.
But if you really think it's better and the best pattern I can create a dedicated file.

Reply:
Yes just to show the good practice I would rather separate, even if I agree here it's a not a real problem, but just for the example :)

Reply:
Ok done